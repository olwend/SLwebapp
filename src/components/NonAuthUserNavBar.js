import React from "react";

import { Link } from "react-router-dom";

import {
  Nav,
  UncontrolledDropdown,
  DropdownMenu,
  DropdownItem,
  DropdownToggle,
} from "reactstrap";

import { PersonIcon } from "@primer/octicons-react";

const NonAuthUserNavBar = () => (
  <Nav navbar>
    <UncontrolledDropdown nav inNavbar>
      <DropdownToggle nav caret data-cy="dropdown-login-menu">
        <PersonIcon />
      </DropdownToggle>
      <DropdownMenu right>
        <DropdownItem tag={Link} to="/login" data-cy="dropdown-login-button">
          Login
        </DropdownItem>
      </DropdownMenu>
    </UncontrolledDropdown>
  </Nav>
);

export default NonAuthUserNavBar;
