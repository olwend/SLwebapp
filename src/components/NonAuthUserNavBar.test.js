import React from 'react'

import { shallow } from 'enzyme'

import NonAuthUserNavBar from './NonAuthUserNavBar'

test('Renders properly', () => expect(shallow(<NonAuthUserNavBar />)).toMatchSnapshot())
