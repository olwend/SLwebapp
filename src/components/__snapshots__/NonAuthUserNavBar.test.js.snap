// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<Nav
  navbar={true}
  tag="ul"
  vertical={false}
>
  <UncontrolledDropdown
    inNavbar={true}
    nav={true}
  >
    <DropdownToggle
      aria-haspopup={true}
      caret={true}
      color="secondary"
      data-cy="dropdown-login-menu"
      nav={true}
    >
      <PersonIcon
        className="octicon"
        size={16}
        verticalAlign="text-bottom"
      />
    </DropdownToggle>
    <DropdownMenu
      flip={true}
      right={true}
      tag="div"
    >
      <DropdownItem
        data-cy="dropdown-login-button"
        tag={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/login"
        toggle={true}
      >
        Login
      </DropdownItem>
    </DropdownMenu>
  </UncontrolledDropdown>
</Nav>
`;
